spec/message.abnf

```diff
@@ -83,5 +83,15 @@ content-char      = %x01-08        ; omit NULL (%x00), HTAB (%x09) and LF (%x0A)
 escaped-char = backslash ( backslash / "{" / "|" / "}" )
 backslash    = %x5C ; U+005C REVERSE SOLIDUS "\"
 
-; Whitespace
-s = 1*( SP / HTAB / CR / LF / %x3000 )
+; Optional whitespace
+owsp = *(s / bidi)
+
+; Required whitespace
+wsp = (owsp) 1*s (owsp)
+
+; Bidirectional marks and isolates
+; ALM / LRM / RLM / LRI, RLI, FSI & PDI
+bidi = %x061C / %x200E / %x200F / %x2066-2069
+
+; Whitespace characters
+s = SP / HTAB / CR / LF / %x3000
```

Comment:
I find replacing `[s]` and `s` with `owsp` and `wsp` to make the ABNF harder to read, as these longer names make it difficult to "see" them as representations of whitespace.

Could we do something like this instead?

```suggestion
; Optional whitespace
_ = *(ws / bidi)

; Required whitespace
s = _ ws _

; Bidirectional marks and isolates
; ALM / LRM / RLM / LRI, RLI, FSI & PDI
bidi = %x061C / %x200E / %x200F / %x2066-2069

; Whitespace characters
ws = SP / HTAB / CR / LF / %x3000
```

Reply:
`_` isn't a valid ABNF rule name:

```abnf
         rulename       =  ALPHA *(ALPHA / DIGIT / "-")
```

Alternatives might be `s` vs. `o` (optional) or perhaps `s` (required) and `s-` (optional)

Reply:
Ah, you're right. I think `s` and `o` is probably the least worst option here. `s-` looks like a typo.

Reply:
Also, to reduce nominal lookahead, I think the required whitespace rule ought to look like this: 

```abnf
s = *bidi ws o
```

Reply:
The `o` would need a `*`, I think:

```abnf
s = *bidi ws *o
```

Reply:
No need, an empty string satisfies `o` as it's got the `*` built into it.